scrapy/downloadermiddlewares/cookies.py

```diff
@@ -71,7 +72,10 @@ def _debug_set_cookie(self, response, spider):
 
     def _format_cookie(self, cookie):
         # build cookie string
-        cookie_str = '%s=%s' % (cookie['name'], cookie['value'])
```

Comment:
This is the part I'm least certain about. The tests were failing for py3 because the `b` char before a `bytes` object was being appended to the string, i.e.: `'%s=%s' % ('e', b'\xc3\xaa')`returned `"e=b'\\xc3\\xaa'"`, making the test comparison fail.
@kmike What do you think about it?